Cython/Includes/cpython/datetime.pxd

```diff
@@ -1,27 +1,94 @@
 from cpython.object cimport PyObject
+from cpython.version cimport PY_VERSION_HEX
 
 cdef extern from "Python.h":
     ctypedef struct PyTypeObject:
         pass
 
 cdef extern from "datetime.h":
     """
-    #if PY_MAJOR_VERSION < 3 && !defined(PyDateTime_DELTA_GET_DAYS)
-    #define PyDateTime_DELTA_GET_DAYS(o)         (((PyDateTime_Delta*)o)->days)
+    /* Backport for Python < 2.7.5 */
+    #if PY_VERSION_HEX < 0x02070500
+        #ifndef PyDateTime_DELTA_GET_DAYS
+            #define PyDateTime_DELTA_GET_DAYS(o) (((PyDateTime_Delta*)o)->days)
+        #endif
+        #ifndef PyDateTime_DELTA_GET_SECONDS
+            #define PyDateTime_DELTA_GET_SECONDS(o) (((PyDateTime_Delta*)o)->seconds)
+        #endif
+        #ifndef PyDateTime_DELTA_GET_MICROSECONDS
+            #define PyDateTime_DELTA_GET_MICROSECONDS(o) (((PyDateTime_Delta*)o)->microseconds)
+        #endif
     #endif
-    #if PY_MAJOR_VERSION < 3 && !defined(PyDateTime_DELTA_GET_SECONDS)
-    #define PyDateTime_DELTA_GET_SECONDS(o)      (((PyDateTime_Delta*)o)->seconds)
+
+    /* Backport for Python < 3.6 */
+    #if PY_VERSION_HEX < 0x030600a4
+        #ifndef PyDateTime_TIME_GET_FOLD
+            #define PyDateTime_TIME_GET_FOLD(o) 0
+        #endif
+        #ifndef PyDateTime_DATE_GET_FOLD
+            #define PyDateTime_DATE_GET_FOLD(o) 0
+        #endif
+    #endif
+
+    /* Backport for Python < 3.6 */
+    #if PY_VERSION_HEX < 0x030600a4
+        #define __Pyx_DateTime_DateTimeWithFold(year, month, day, hour, minute, second, microsecond, tz, fold) \
+            PyDateTimeAPI->DateTime_FromDateAndTime(year, month, day, hour, minute, second, \
+                microsecond, tz, PyDateTimeAPI->DateTimeType)
+        #define __Pyx_DateTime_TimeWithFold(hour, minute, second, usecond, tz, fold) \
+            PyDateTimeAPI->Time_FromTime(hour, minute, second, usecond, tz, PyDateTimeAPI->TimeType)
+    #else /* For Python 3.6+ so that we can pass tz */
+        #define __Pyx_PyDateTime_FromDateAndTimeAndFold(hour, minute, second, usecond, tz, fold) \
+            PyDateTimeAPI->DateTime_FromDateAndTimeAndFold(year, month, day, hour, minute, second, \
+                microsecond, tz, fold, PyDateTimeAPI->DateTimeType)
+        #define __Pyx_DateTime_TimeWithFold(hour, minute, second, usecond, tz, fold) \
+            PyDateTimeAPI->Time_FromTimeAndFold(hour, minute, second, usecond, tz, fold, PyDateTimeAPI->TimeType)
     #endif
```

Comment:
These don't declare the same functions. We must declare the names here that we use in the inline functions below.